spec/models/approval_spec.rb

```diff
@@ -20,4 +20,86 @@
       expect(approval.api_token).to eq(nil)
     end
   end
+
+  describe '#on_approved_entry' do
+    it "notified the proposal if the root gets approved" do
+      expect(approval.proposal.approved?).to eq false
+      approval.make_actionable!
+      approval.approve!
+      expect(approval.proposal.approved?).to eq true
+    end
+
+    it "does not notify the proposal if a child gets approved" do
+      proposal = FactoryGirl.create(:proposal)
+      proposal.root_approval = Approvals::Parallel.new
+      child1 = proposal.add_approver("child1@agency.gov")
+      proposal.add_approver("child2@agency.gov")
+      proposal.root_approval.make_actionable!
+
+      expect(approval.proposal.approved?).to eq false
+      child1.reload.approve!
+      expect(approval.proposal.approved?).to eq false
+      expect(approval.proposal.pending?).to eq true
+    end
+  end
+
+  describe "complicated approval chains" do
+    # Approval hierarchy version of needing two of the following:
+    # 1) Amy AND Bob
+    # 2) Carrie
+    # 3) Dan THEN Erin
```

Comment:
This made me think: now that we have this more...flexible...structure, we should have a more expressive way to set up the approval trees, that looks more like this comment than the huge block below. Not a blocker.